### ISSUE TITLE
Double names should be used in expectation failures

### ISSUE THREAD

**Issue by myronmarston** [rspec/rspec-expectations#688]
``` ruby
    example do
      foo = double("Foo")
      bar = double("Bar")

      expect([foo]).to include(bar)
    end
```

Fails with:

```
     Failure/Error: expect([foo]).to include(bar)
       expected [#<RSpec::Mocks::Double:0x84433530 @name="Foo">] to include #<RSpec::Mocks::Double:0x844333c8 @name="Bar">
       Diff:
       @@ -1,2 +1,2 @@
       -[#<RSpec::Mocks::Double:0x844333c8 @name="Bar">]
       +[#<RSpec::Mocks::Double:0x84433530 @name="Foo">]
```

...which is pretty ugly.  This would be so much nicer:

```
     Failure/Error: expect([foo]).to include(bar)
       expected [(Double "Foo")] to include (Double "Bar")
       Diff:
       @@ -1,2 +1,2 @@
       -[(Double "Bar")]
       +[(Double "Foo")]
```

Care will have to be taken to maintain the independence of rspec-mocks and rspec-expectations. I suspect that this may need changes in rspec-support to support this. (Or perhaps some `if defined?(RSpec::Mocks)` checks).

**Comment by sivagollapalli** [comment]
@myronmarston I think we can achieve this if we modify `inspect` method on `lib/rspec/mocks/test_double.rb` as `Double (#{@name})`. Please check https://github.com/sivagollapalli/rspec-mocks/commit/42414afc092ed765570c457972f913a2ae460cf6. If it is fine I will send a PR .

**Comment by myronmarston** [comment]
I think that's a reasonable solution, although I noticed that the existing `inspect` and `to_s` definitions are obviously intentional, to the point of having a spec for it:

https://github.com/rspec/rspec-mocks/blob/v3.2.0/lib/rspec/mocks/test_double.rb#L41-L48

https://github.com/rspec/rspec-mocks/blob/v3.2.0/spec/rspec/mocks/double_spec.rb#L20-L23

That predates my involvement with RSpec, and I'm not sure the reason that was chosen for `inspect` rather than something like `"Double (#{@name})"` like you suggest.

@dchelimsky, do you know why `RSpec::Mocks::Double#inspect` was chosen to be that way in RSpec 1?

**Comment by sivagollapalli** [comment]
@myronmarston Shall I make a PR to `rspec-mocks` with above change?

**Comment by myronmarston** [comment]
Sorry for the delay, @sivagollapalli.  Changing `Double#inspect` may have undesirable side effects.  We're moving in the direction of centralizing object inspection for the purpose of failure output in rspec-support (see rspec/rspec-support#184), so maybe you can work with @gavingmiller to get support for doubles added to the new `RSpec::Support::ObjectInspector`?

**Comment by sivagollapalli** [comment]
@myronmarston I made a PR(https://github.com/rspec/rspec-support/pull/190) with the change as discussed above. Please check

**Comment by dchelimsky** [comment]
I don't remember the motivation, and looking back that example was in place when we split the repos (a64df194cccf5acb9dbdc0b18154d7a1ceebfc37) in 2009, so git history won't help us (the referenced commit doesn't exist in the rspec repo). Sorry I can't help.

**Comment by JonRowe** [comment]
So did #190 close this?

**Comment by myronmarston** [comment]
No, rspec/rspec-mocks#920 needs to be finished first.

**Comment by myronmarston** [comment]
Closing in favor of rspec/rspec-mocks#932.
